### PR TITLE
BaSP-TG-06: Create Employee component.

### PR DESCRIPTION
**Pull request by JulianFloresDev**
Development of the necessary components to implement a complete CRUD on the "Employees" entity.

> Show Resource List
> Edit Button
> Edit feature
> Delete button
> Delete feature
> Succes message
> Error message
> Redirect button to Add feature

### CODE REVIEW

File: src/Components/Employees/Input/index.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import styles from './input.module.css';
+
+const Input = ({ filterEmployees }) => {
+  return (
+    <div className={styles.container}>
+      <label htmlFor="input-search">
+        <span className="material-symbols-outlined icon-search" id="icon-sreach">
```
Comment (kolodani) [inline src/Components/Employees/Input/index.js:7]: typo error: search not sreach

File: src/Components/Employees/Modal/index.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+          <button
+            type="button"
+            className={styles.button}
+            onClick={() => {
+              closeModal();
+            }}
+          >
+            Close
+          </button>
+          {/* <button type="button" className={styles.button} onClick={() => setModalState(false)}>
```
Comment (kolodani) [inline src/Components/Employees/Modal/index.js:37]: why you have this code commented

File: src/Components/Employees/Table/index.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+              </td>
+              <td>{element.email}</td>
+              <td>{element.phone}</td>
+              <td>{element.location}</td>
+              <td className={styles.buttons}>
+                <button className={styles.editBtn} onClick={() => editEmployee(element)}>
+                  Edit
+                </button>
+                <button className={styles.deleteBtn} onClick={() => deleteEmployee(element._id)}>
+                  x
```
Comment (kolodani) [inline src/Components/Employees/Table/index.js:36]: i think is better use here the were 'delete' word instead of 'x'

**Review by kolodani — APPROVED** [review]
well done, nice work!

File: src/Components/Employees/Modal/index.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import styles from './modal.module.css';
+
+const Modal = ({ modalData, modalState, setModalState, render }) => {
+  console.log(modalData.message);
```
Comment (Alfondalix) [inline src/Components/Employees/Modal/index.js:4]: We should delete this line

File: src/Components/Employees/index.js [hunk 5]
```diff
@@ -1,25 +1,100 @@
...
+  const getAllEmployees = async () => {
+    try {
+      await fetch(`${process.env.REACT_APP_API_URL}/employees`)
+        .then((response) => response.json())
+        .then((response) => {
+          saveEmployees(response.data || []);
+          saveFilteredEmployees(response.data || []);
+        });
+    } catch (error) {
+      console.log(error);
```
Comment (Alfondalix) [inline src/Components/Employees/index.js:24]: We could change it to console.error(error)

File: src/Components/Employees/index.js [hunk 6]
```diff
@@ -1,25 +1,100 @@
 import { useEffect, useState } from 'react';
 import styles from './employees.module.css';
+import Table from './Table';
+import Input from './Input';
+// import EditEmployeeModal from './Edit';
```
Comment (Alfondalix) [inline src/Components/Employees/index.js:5]: We should delete this line if we're not using it

**Review by Alfondalix — CHANGES_REQUESTED** [review]

File: src/Components/Employees/Modal/index.js (hunk 2 above)
Comment (JulianFloresDev) [reply src/Components/Employees/Modal/index.js:37]: Resolve

File: src/Components/Employees/Table/index.js (hunk 3 above)
Comment (JulianFloresDev) [reply src/Components/Employees/Table/index.js:36]: I think I like this short way.

File: src/Components/Employees/index.js (hunk 5 above)
Comment (JulianFloresDev) [reply src/Components/Employees/index.js]: Resolve

File: src/Components/Employees/index.js (hunk 6 above)
Comment (JulianFloresDev) [reply src/Components/Employees/index.js]: Resolve

File: src/Components/Employees/Form/index.js [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+            >
+              {buttonMsg}
+            </button>
+            <button
+              className={styles.closeBtn}
+              onClick={(e) => {
+                e.preventDefault();
+                render(0);
+                // changeState(!modalState);
+                // closeModal();
```
Comment (FacuDLF) [inline src/Components/Employees/Form/index.js:136]: Commented lines should be deleted.

Comment (JulianFloresDev) [reply src/Components/Employees/Form/index.js]: Changes done

**Review by FacuDLF — APPROVED** [review]

**Review by agusgatto8 — APPROVED** [review]
Done

**Review by agusgatto8 — APPROVED** [review]

**Review by kolodani — APPROVED** [review]
well done!!!

**Review by Alfondalix — APPROVED** [review]

**Pull request merged**